modules/tinymce/src/core/test/ts/browser/WindowManagerTest.ts

```diff
@@ -41,4 +47,66 @@ describe('browser.tinymce.core.WindowManagerTest', () => {
 
     editor.off('CloseWindow OpenWindow');
   });
+
+  context('WindowManager dialog validation workaround', () => {
+    const hook = TinyHooks.bddSetup<Editor>({
+      base_url: '/project/tinymce/js/tinymce',
+      plugins: 'link',
+      toolbar: 'link',
+      setup: (editor: Editor) => {
+        editor.on('PreInit', () => {
+          const originalWindowManager = editor.windowManager;
+          editor.windowManager = Tools.extend({}, originalWindowManager, {
+            open: (spec: Dialog.DialogSpec<LinkDialogData>) => {
+              if (spec.title === 'Insert/Edit Link') {
+                const newSpec = Tools.extend({}, spec, {
+                  onChange: (api: Dialog.DialogInstanceApi<LinkDialogData>, details: Dialog.DialogChangeDetails<LinkDialogData>) => {
+                    spec.onChange(api, details);
+                    if (details.name === 'url' || details.name === 'link' || details.name === 'anchor') {
+                      const data = api.getData();
+                      if (data.url.value.length === 0) {
+                        api.disable('save');
+                      } else {
+                        api.enable('save');
+                      }
+                    }
+                  }
+                });
+                const api = originalWindowManager.open(newSpec);
+                if (spec.initialData.url.value.length === 0) {
+                  api.disable('save');
+                }
+
+                return api;
+              } else {
+                return originalWindowManager.open(spec);
+              }
+            }
+          });
+        });
+      }
+    }, [ Theme, LinkPlugin ]);
+
+    it('TINY-7738: Regression test for supported dialog validation workaround', async () => {
```

Comment:
I think if we are going to have a test in this file, it might be more suitable to just have a basic spec with a URL input and save button rather than importing and using the link plugin dialog.

Reply:
Agree, will do

Reply:
Another option is to just move this test to the link plugin, as remember we are trying to make sure this exact case works (ie we also need to make sure the ui component names don't change)